packages/codd/tests/test_codd_cli.py

```
@@ -651,6 +717,63 @@ def test_path_in_scope_single_star_is_segment_aware() -> None:
+
+
+def test_scope_pattern_unterminated_bracket_is_literal() -> None:
+    # 閉じ ] が無い場合は fnmatch と同様リテラル [ として扱う（クラッシュしない）。
+    regex = cli._scope_pattern_to_regex("docs/[abc.md")
+    assert regex.fullmatch("docs/[abc.md") is not None
+    assert regex.fullmatch("docs/a.md") is None
+
+
+def test_scope_pattern_invalid_char_range_is_safe_non_match() -> None:
+    # `[z-a]` は逆順の不正な範囲。re.compile が re.error で落ちるため、Path.glob
+    # （fnmatch）と同様「常に非マッチ」として安全に扱う（クラッシュしない。
```

Ruff (0.15.21)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)